libs/angular/src/lib/modal/modal.component.ts

```diff
@@ -0,0 +1,130 @@
+import { ChangeDetectionStrategy, Component, ElementRef, EventEmitter, HostBinding, Input, OnDestroy, Output, ViewChild } from '@angular/core';
+import { ModalType, Size } from '@sebgroup/extract';
+import { NggModalService } from './modal.service';
+
+@Component({
+    selector: 'ngg-modal',
+    styleUrls: ['./modal.component.scss'],
+    templateUrl: './modal.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NggModalComponent implements OnDestroy {
+    @Input() public modalType?: ModalType
+    @Input() public header?: string
+    @Input() public confirm?: string
+    @Input() public dismiss?: string
+    @Input() public size?: Size
+
+    @Input() 
+    public get isOpen(): boolean | undefined {
+        return this._isOpen;
+    }
+    public set isOpen(value: boolean | undefined) {
+        this._isOpen = value;
+
+        if (value) {
+            this.modalService.lockBody(this.ref.nativeElement);
+        } else {
+            this.modalService.releaseBody(this.ref.nativeElement);
+        }
+    }
+
+    @Output() public isOpenChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+    @Output() public onClose: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
+    @Output() public onConfirm: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
+    @Output() public onDismiss: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();
+    
+    @HostBinding('class.open') get open() { return this.isOpen; }
+    @ViewChild('backdrop') private backdropRef?: ElementRef<HTMLInputElement>;
+    private _isOpen?: boolean;
+    
+    constructor(private ref: ElementRef<HTMLElement>, private modalService: NggModalService) {
+    }
+    
+    public handleCloseClick(event: MouseEvent) {
+        this.closeModal(event);
+    }
+
+    public handleBackdropClick(event: MouseEvent) {
+        if (event.target == this.backdropRef?.nativeElement)
+            this.closeModal(event);
+    }
+    
+    public handleDismiss(event: MouseEvent) {
+        if (this.onDismiss.observers.length > 0)
```

Comment:
is there a reason you only emit if there is an observer?

Reply:
For `onDismiss` and `onConfirm` no, but for `onClose` yes. For `onClose` its to be able to intercept the closing of the modal outside of the component, where you can have some logic determine if you want to close the modal or not by updating the `isOpen` property. For example to show a confirmation dialog (Are you sure you want to close? Yes or No) when you close (e.g. press the close/cross button) the modal.

But for `onDismiss` and `onConfirm` its not necessary to only emit if there are any observers, but I guess you would only want to emit if you have any observers? But perhaps that check is already done in emit?

Reply:
Gotcha! :) 

Yes, if the emitter doesn't have any registered observers it will simply not call any. So there is no need to check the amount of observers here, expect for `onClose` then 👍